db/migrate/20170609132731_add_fedora4_record_identifiers.rb

```diff
@@ -0,0 +1,19 @@
+class AddFedora4RecordIdentifiers < ActiveRecord::Migration
+
+  def up
+    say_with_time("Add Fedora4 record identifers") do
+      MediaObject.find_each({},{batch_size:5}) do |mo|
+        say("MediaObject: #{mo.id}, URI: #{mo.uri}")
+        if mo.descMetadata.record_identifier.empty?
+          mo.descMetadata.ensure_identifier_exists! mo.uri
```

Comment:
Do you really need this?  Wouldn't a mo.save be sufficient to trigger the callback which runs ensure_identifier_exists!?